lib/ipam/ipam.go

```diff
@@ -773,8 +792,15 @@ func (c ipamClient) ClaimAffinity(ctx context.Context, cidr net.IPNet, host stri
 // its affinity will not be released and no error will be returned.
 // If an empty string is passed as the host, then the hostname is automatically detected.
 func (c ipamClient) ReleaseAffinity(ctx context.Context, cidr net.IPNet, host string) error {
+	// Verify the requested CIDR falls within a configured pool.
+	pool := c.blockReaderWriter.withinConfiguredPools(net.IP{IP: cidr.IP})
```

Comment:
I think we should rename this function from `withinConfiguredPools` to something like `getPoolForIP` or something like that to match its new behavior.